nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+        example = [ "postmaster@sympa.example.org" ];
+        description = ''
+          The list of the email addresses of the listmasters
+          (users authorized to perform global server commands).
+        '';
+      };
+
+      mainDomain = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "lists.example.org";
+        description = ''
+          Main domain to be used in sympa.conf.
+          If null, one of the <option>domains</option> is chosen for you.
+        '';
+      };
+
+      domains = mkOption {
+        type = types.attrsOf (types.submodule {
+          options = {
+            webHost = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "archive.example.org";
+              description = ''
+                Domain part of the web interface URL (no web interface for this domain if null).
+                DNS record of type A (or AAAA) has to exist with this value.
+              '';
+            };
+            webLocation = mkOption {
+              type = types.str;
+              default = "/";
+              example = "/sympa";
+              description = "URL path part of the web interface.";
+            };
+            extraConfig = mkOption {
+              type = types.lines;
+              default = "";
+              description = ''
+                These lines go to the end of the robot.conf verbatim.
+              '';
+            };
+          };
+        });
+        description = ''
+          Email domains handled by this instance. There have
+          to be MX records for keys of this attribute set.
+        '';
+        example = literalExample ''
+          {
+            "lists.example.org" = {
+              webHost = "lists.example.org";
+              webLocation = "/";
+            };
+            "sympa.example.com" = {
+              webHost = "example.com";
+              webLocation = "/sympa";
+            };
+          };
+        '';
+      };
+
+      database = {
+        type = mkOption {
+          type = types.enum [ "SQLite" "PostgreSQL" "MySQL" ];
+          default = "SQLite";
+          example = "MySQL";
+          description = "Database engine to use.";
+        };
+
+        host = mkOption {
+          type = types.nullOr types.str;
+          default = "127.0.0.1";
+          description = "Database host address.";
+        };
+
+        port = mkOption {
+          type = types.nullOr types.port;
+          default = null;
+          description = "Database port.";
+        };
+
+        name = mkOption {
+          type = types.str;
+          default = if cfg.database.type == "SQLite" then "${dataDir}/sympa.sqlite" else "sympa";
+          description = ''
+            Database name. When using SQLite this must be an absolute
+            path to the database file.
+          '';
+        };
+
+        user = mkOption {
+          type = types.nullOr types.str;
```

Comment:
I would check the database type and either set the db user to null for sqlite otherwise to the same name as the sympa user as a default

Reply:
Can we just use `cfg.user` by default instead? The `SQLite` module seems to ignore the value.